File: app/src/main/java/com/beekay/thoughts/model/Thought.java
package com.beekay.thoughts.model;

import java.io.Serializable;

public class Thought implements Serializable {

    private Long id;
    private String thoughtText;
    private String timestamp;
    private String imgSource;
    private byte[] img;
    private boolean starred;

    public Thought() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getThoughtText() {
        return thoughtText;
    }

    public void setThoughtText(String thoughtText) {
        this.thoughtText = thoughtText;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public String getImgSource() {
        return imgSource;
    }

    public void setImgSource(String imgSource) {
        this.imgSource = imgSource;
    }

    public byte[] getImg() {
        return img;
    }

    public void setImg(byte[] img) {
        this.img = img;
    }

    public boolean isStarred() {
        return starred;
    }

    public void setStarred(boolean starred) {
        this.starred = starred;
    }
}

File: app/src/main/java/com/beekay/thoughts/model/Thought.kt
package com.beekay.thoughts.model

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

/**
 * Created by Krishna by 15-11-2020
 */
private val dateFormat = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm:ss")
@Entity(tableName = "thought")
data class Thought(
        @PrimaryKey(autoGenerate = true)
        @ColumnInfo(name = "tid")
        @NonNull
        val id: Long = 0,

        @ColumnInfo(name = "thought_text")
        @NonNull
        val thought: String,

        @ColumnInfo(name = "createdon")
        @NonNull
        val createdOn: String,

        @ColumnInfo(name = "updatedon")
        @NonNull
        val updatedOn: String = LocalDateTime.now().format(dateFormat),

        @ColumnInfo(name = "img_src")
        val imgSource: String?,

        @ColumnInfo(name = "starred")
        @NonNull
        var starred: Boolean = false
)